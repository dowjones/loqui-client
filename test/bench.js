
var Benchmark = require('benchmark')
var suite = new Benchmark.Suite

var winston = require('winston')
winston.add(winston.transports.File, { filename: 'logs.log' });

var loqui = require('../index').createClient()
var loquiQ = require('../index').createClient({ queueSize: 2 })
var log = []

// add tests
suite.add('Winston', function() {
  winston.log('info', 'Hello distributed log files!')
})
.add('Loqui', function() {
  loqui.log('info', 'Hello distributed log files!')
})
.add('Loqui-QueueSize-50%', function() {
  loquiQ.log('info', 'Hello distributed log files!')
})

// add listeners
.on('cycle', function(event) {
  log.push(String(event.target))
})
.on('complete', function() {
  console.log(log)
  console.log('Fastest is ' + this.filter('fastest').pluck('name'))
})
// run async
.run({ 'async': true })