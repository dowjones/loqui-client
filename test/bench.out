[ 'Loqui x 66,597 ops/sec ±45.47% (28 runs sampled)',
  'Winston x 17,384 ops/sec ±13.97% (58 runs sampled)',
  'Loqui-Queued x 104,315 ops/sec ±24.44% (43 runs sampled)' ]